src/transformers/pipelines/__init__.py

```diff
@@ -621,11 +621,14 @@ def pipeline(
                     import kenlm  # to trigger `ImportError` if not installed
                     from pyctcdecode import BeamSearchDecoderCTC
 
-                    language_model_glob = os.path.join(BeamSearchDecoderCTC._LANGUAGE_MODEL_SERIALIZED_DIRECTORY, "*")
-                    alphabet_filename = BeamSearchDecoderCTC._ALPHABET_SERIALIZED_FILENAME
-                    allow_regex = [language_model_glob, alphabet_filename]
+                    if os.path.isdir(model_name):
```

Comment:
```suggestion
                    if os.path.isdir(model_name) or os.path.isfile(model_name):
```
In this case I think `BeamSearchDecoderCTC.load_from_dir` will fail gracefully, so it makes sense to include it here IMO :-)

Reply:
Wondering if then we should include the same check here: https://github.com/huggingface/transformers/blob/7732d0fe7a759c9844215920e9f1c5540eafb1a6/src/transformers/models/wav2vec2_with_lm/processing_wav2vec2_with_lm.py#L130

Reply:
Agree with @versae both code path should agree at least.

Reply:
Good point @versae - do you mind adapting it there as well? :-)

Reply:
Done.